socrates/lib/registration/index.js

```diff
@@ -21,22 +21,37 @@ var roomOptions = beans.get('roomOptions');
 
 var app = misc.expressAppIn(__dirname);
 
-function isRegistrationOpen() {
-  // we currently set this to false on production system, because this feature is still in development
-  // the conf variable is used as a backdoor for testing
-  return app.get('env') !== 'production' && !conf.get('registrationIsClosed');
+function isRegistrationOpen(registrationParam) {
+  return moment(conf.get('registrationOpensAt')).isBefore(moment())
+    || (registrationParam && registrationParam === conf.get('registrationParam'));
+}
+
+function registrationOpensIn() {
+  var registrationOpening = moment(conf.get('registrationOpensAt'));
+  var reference = moment();
+  if (registrationOpening.isAfter(reference)) {
+    var inDays = registrationOpening.diff(reference, 'days');
+    var inHours = registrationOpening.diff(reference.add(inDays, 'days'), 'hours');
+    var inMinutes = registrationOpening.diff(reference.add(inHours, 'hours'), 'minutes');
+    return {days: inDays, hours: inHours, minutes: inMinutes};
+  }
+  return undefined;
 }
 
 app.get('/', function (req, res, next) {
   activitiesService.getActivityWithGroupAndParticipants(socratesConstants.currentUrl, function (err, activity) {
     if (err || !activity) { return next(err); }
     var options = roomOptions.all(activity, res.locals.accessrights.memberId(), isRegistrationOpen());
+
     res.render('get', {
```

Comment:
Du meinst, in ein Objekt zusammenfassen? Halte ich nicht für sehr hilfreich...
Oder meinst Du logisch zusammenfassen? Ich wüsste nicht wie (außer durch Verlagern der Berechnungslogik ins Jade - grusel...)